### PR TITLE
Rework entities not to hold defs

### PR DESCRIPTION
**Pull request by BWalusiak**

### CODE REVIEW

File: core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java [hunk 1]
```diff
@@ -12,13 +12,18 @@
 public class PlayerEntity extends Entity {
     private static final float ANIMATION_DURATION = 0.2f;
 
+
+    private final Inventory inventory;
+    private float dragModifier;
+    private float maxSpeedModifier;
+    private int textureVariant;
```
Comment (RouNNdeL) [inline core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java]: Texture variant should probably be final

File: core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java [hunk 2]
```diff
@@ -27,31 +32,34 @@ public PlayerEntity(World world, PlayerDef def) {
...
 
     public float getMaxSpeedModifier() {
-        return def.maxSpeedModifier * def.inventory.getMovementSpeedMultiplier();
+        return maxSpeedModifier * inventory.getMovementSpeedMultiplier();
     }
 
-    public void setMaxSpeedModifier(float maxSpeedModifier) {
-        def.maxSpeedModifier = maxSpeedModifier;
+    public void setMaxSpeedModifier(float newMaxSpeedModifier) {
+        maxSpeedModifier = newMaxSpeedModifier;
```
Comment (RouNNdeL) [inline core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java]: Rename to match other setters and use this.maxSpeedModifer to assign

File: core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java [hunk 3]
```diff
@@ -27,31 +32,34 @@ public PlayerEntity(World world, PlayerDef def) {
...
 
     public float getDragModifier() {
-        return def.dragModifier;
+        return dragModifier;
     }
 
-    public void setDragModifier(float dragModifier) {
-        def.dragModifier = dragModifier;
+    public void setDragModifier(float newDragModifier) {
+        dragModifier = newDragModifier;
```
Comment (RouNNdeL) [inline core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java]: Rename to match other setters and use this.dragModifire to assign

File: core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java (hunk 1 above)
Comment (BWalusiak) [reply core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java]: Fixed

File: core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java (hunk 2 above)
Comment (BWalusiak) [reply core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java]: Fixed

File: core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java (hunk 3 above)
Comment (BWalusiak) [reply core/src/main/java/com/cyberbot/bomberman/core/models/entities/PlayerEntity.java]: Fixed

**Review by RouNNdeL — APPROVED** [review]

**Pull request merged**